openlibrary/components/BarcodeScanner/components/OLBarcodeScanner.vue

```diff
@@ -0,0 +1,191 @@
+        },
+    },
+    data() {
+        return {
+            returnTo: new URLSearchParams(location.search).get('returnTo'),
+            BOX_STYLE: {color: 'green', lineWidth: 2},
+            RESULT_BOX_STYLE: {color: 'blue', lineWidth: 2},
+            RESULT_LINE_STYLE: {color: 'red', lineWidth: 15},
+            submitISBNThrottled: new ThrottleGrouping({
+                func: this.submitISBN.bind(this),
+                // Use the most frequent
+                reducer: (groupOfArgs) => {
+                    const isbnCounts = Array.from(
+                        Object.entries(
+                            countBy(groupOfArgs, (arg) => arg[0])
+                        )
+                    );
+
+                    /* eslint-disable no-unused-vars */
+                    const mostFrequentISBN = maxBy(isbnCounts, ([isbn, count]) => count)[0];
+                    return groupOfArgs.reverse().find((args) => args[0] === mostFrequentISBN);
+                },
+                wait: 300,
+            }).asFunction(),
+            childState: this.state,
+            childLastISBN: this.lastISBN,
+        }
+    },
+    methods: {
+        start() {
+            Quagga.init({
+                locator: {
+                    halfSample: true,
+                },
+                inputStream: {
+                    name: 'Live',
+                    type: 'LiveStream',
+                    target: document.getElementById('interactive'),
+                    constraints: {
+                    // Vertical - This is *essential* for iPhone/iPad
+                        aspectRatio: {ideal: 720/1280},
+                    },
+                },
+                decoder: {
+                    readers: ['ean_reader']
+                },
+            }, async (err) => {
+                if (err) throw err;
+
+                const track = Quagga.CameraAccess.getActiveTrack();
+                if (track && typeof track.getCapabilities === 'function') {
+                    const capabilities = track.getCapabilities();
+                    // Use a higher resolution
+                    if (capabilities.width.max >= 1280 && capabilities.height.max >= 720) {
+                        await track.applyConstraints({advanced: [{width: 1280, height: 720}]});
+                    }
+                }
+
+                const quaggaVideo = document.getElementById('interactive').getElementsByTagName('video')[0];
+                const readISBNButton = /** @type {HTMLButtonElement} */(document.querySelector('.barcodescanner__read-isbn'));
+
+                const ocrScanner = new OCRScanner();
+                // document.body.append(s.userCanvas);
+                ocrScanner.onISBNDetected((isbn) => {
+                    this.submitISBNThrottled(isbn, '/static/images/openlibrary-logo-tighter.svg');
+                });
+                ocrScanner.init();
+
+                readISBNButton.addEventListener('click', async () => {
```

Comment:
These should be handled with vue-style click handlers, e.g. 

```
<button @click="handleIsbnDetected>
```

Another reason why dissolving this component into the parent will make things a little easier!